src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -0,0 +1,184 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.Util;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import java.io.IOException;
+import java.util.List;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.github.GHApp;
+import org.kohsuke.github.GHAppInstallation;
+import org.kohsuke.github.GHAppInstallationToken;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GitHubBuilder;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+import static org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator.DescriptorImpl.getPossibleApiUriItems;
+import static org.jenkinsci.plugins.github_branch_source.JwtHelper.createJWT;
+
+public class GitHubAppCredentials extends BaseStandardCredentials implements StandardUsernamePasswordCredentials {
```

Comment:
Since the user name isn't needed could this implement a different interface that doesn't include user name?

Reply:
If it implements something which does include a user name, then the git plugin (and other plugins that require a username / password credential) will need to be extended to support that different credential type.  By choosing to implement `StandardUsernamePasswordCredentials`, the credential consumers  can use it without requiring that they be changed for a new credential type.

I think that I've described this case correctly, though @timja may need to correct my misunderstanding.

Reply:
Mark you are absolutely correct, the git plugin is the reason to implement this interface